src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua

```diff
@@ -278,3 +278,76 @@ function drawImage(image, xPos, yPos)
         end
     end
 end
+
+--- Parses an nft image from a multi-line string
+--
+-- @tparam string image The string containing the raw-nft-image data.
+-- @treturn table The parsed image data, suitable for use with
+-- @{paintutils.drawTextImage}.
+function parseTextImage(image)
+    expect(1, image, "string")
+    local nLine = 1
+    local tImage = {}
+    local sMode = "text"
+    local sBackground = "F"
+    local sForeground = "0"
+    for c in image:gmatch"." do
+        if c == "\n" then
+            nLine = nLine + 1
+        elseif c:byte() == 30 then
+            sMode = "background"
+        elseif c:byte() == 31 then
+            sMode = "foreground"
+        elseif sMode == "background" then
+            sBackground = c
+            sMode = "text"
+        elseif sMode == "foreground" then
+            sForeground = c
+            sMode = "text"
+        elseif sMode == "text" then
+            if tImage[nLine] == nil then
+                tImage[nLine] = {}
+                tImage[nLine]["foreground"] = ""
+                tImage[nLine]["background"] = ""
+                tImage[nLine]["text"] = ""
+            end
+            tImage[nLine]["foreground"] = tImage[nLine]["foreground"]..sForeground
+            tImage[nLine]["background"] = tImage[nLine]["background"]..sBackground
+            tImage[nLine]["text"] = tImage[nLine]["text"]..c
+        end
+    end
+    return tImage
+end
+
+--- Loads an nft image from a file.
+--
+-- @tparam string path The file to load.
+--
+-- @treturn table|nil The parsed image data, suitable for use with
+-- @{paintutils.drawTextImage}, or `nil` if the file does not exist.
+function loadTextImage(path)
+    expect(1, path, "string")
+
+    if fs.exists(path) then
+        local file = io.open(path, "r")
```

Comment:
Instead of doing `fs.exists`, I'd just do:

```lua
local file = io.open
if not file then return nil end
```

Otherwise this will error on directories.